SwiftGit2/Diffs.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  Diffs.swift
+//  SwiftGit2
+//
+//  Created by Jake Van Alstyne on 8/20/17.
+//  Copyright © 2017 GitHub, Inc. All rights reserved.
+//
+
+public struct GitDiffFile {
+	public var oid: OID
+	public var path: String
+	public var size: Int64
+	public var flags: UInt32
+}
+
+public enum GitStatus: Int {
+	case current                = 0
+	case indexNew               = 1
+	case indexModified          = 2
+	case indexDeleted           = 4
+	case indexRenamed           = 8
+	case indexTypeChange        = 16
+	case workTreeNew            = 32
+	case workTreeModified       = 64
+	case workTreeDeleted        = 128
+	case workTreeTypeChange     = 256
+	case workTreeRenamed        = 512
+	case workTreeUnreadable     = 1024
+	case ignored                = 2048
+	case conflicted             = 4096
+
+	public var value: UInt32 {
+		return UInt32(self.rawValue)
+	}
+}
+
+public struct GitDiffDelta {
+	public var status: GitStatus?
+	public var flags: UInt32?
+	public var oldFile: GitDiffFile?
+	public var newFile: GitDiffFile?
+}
+
+public enum GitDiffFlag: Int {
```

Comment:
This looks like it could be an `OptionSet`.

Reply:
Brilliant. Didn't know about optionset. Thanks. Partially doing this to get better at swift, definitely appreciate the feedback.